src/core/qgsdistancearea.h

```diff
@@ -19,6 +19,7 @@
 #include "qgis_core.h"
 #include <QList>
 #include <QReadWriteLock>
+#include <QUuid>
```

Comment:
Why not put a breakpoint in at all codepaths where the ellipsoid parameters are set in QgsDistanceArea and see what's changing them?